sw/device/tests/penetrationtests/firmware/lib/pentest_lib.c

```diff
@@ -369,10 +569,29 @@ status_t pentest_configure_cpu(
     *icache_en = true;
   }
 
+  uint32_t cpuctrl_csr = 0;
   // Enable/disable dummy instructions.
   if (enable_dummy_instr) {
     // Set CSR_REG_CPUCTRL[2].
     CSR_SET_BITS(CSR_REG_CPUCTRL, kCpuctrlDummyInstrEnMask);
+
+    cpuctrl_csr = bitfield_field32_write(
+        cpuctrl_csr, (bitfield_field32_t){.mask = 0x1, .index = 2}, 1);
+    // 0 Dummy instruction every 0 - 4 real instructions
+    // 1 Dummy instruction every 0 - 8 real instructions
+    // 3 Dummy instruction every 0 - 16 real instructions
+    // 7 Dummy instruction every 0 - 32 real instructions
+    cpuctrl_csr = bitfield_field32_write(
+        cpuctrl_csr, (bitfield_field32_t){.mask = 0x7, .index = 3},
+        dummy_instr_count);
```

Comment:
I would recommend switching to:
```suggestion

    cpuctrl_csr = bitfield_field32_write(
        cpuctrl_csr, (bitfield_field32_t){.mask = kCpuctrlDummyInstrMaskMask, .index = kCpuctrlDummyInstrMaskIdx},
        dummy_instr_count);
```
and define these two constants here:

https://github.com/nasahlpa/opentitan/blob/21810c753659dc70fcb2b449704880fdf934c3f8/sw/device/tests/penetrationtests/firmware/lib/pentest_lib.c#L95-L101